File: app/src/main/java/com/damhoe/fieldlines/domain/StartPoint.java
package com.damhoe.fieldlines.domain;

import com.damhoe.fieldlines.domain.Charge;

/**
 * Created by damian on 02.12.2017.
 */
public class StartPoint {
    public double x;
    public double y;
    public Charge center; // determines the direction for calculation (+ /-)

    public StartPoint(double x, double y, Charge center){
        this.x = x;
        this.y = y;
        this.center = center;
    }
}

File: app/src/main/java/com/damhoe/fieldlines/field/domain/StartPoint.kt
package com.damhoe.fieldlines.field.domain

import com.damhoe.fieldlines.app.Vector

/**
 * Each start point is located near a charge.
 * The direction is saved so that the field line only needs
 * to be drawn in one direction.
 */
data class StartPoint(var coordinates: Vector, var direction: Double)